src/PerfView/PerfViewData.cs

```diff
@@ -5713,6 +5713,9 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                 var heapParser = new HeapTraceProviderTraceEventParser(eventSource);
                 Dictionary<Address, StackSourceSample> lastHeapAllocs = null;
 
+                Dictionary<TraceModuleFile, List<HeapAllocationSite>> allocationSites = null;
```

Comment:
Making that returned value a List is inefficient (you end up scanning it later on).   There seems to be no reason for it to not be a Dictionary<uint, string> (mapping the RVA to the type name),   Simply add all the entries returned from  GetHeapAllocationSites  to that dictionary).

Reply:
yes, Dictionary<uint, string> is all I need here. I started with a list because I wasn't getting direct matches on rva until I figured out that the sym.length needed to be added in.